src/main/java/com/intuit/graphql/orchestrator/schema/transform/FederationTransformerPostMerge.java

```diff
@@ -44,11 +44,13 @@ public class FederationTransformerPostMerge implements Transformer<XtextGraph, X
 
   @Override
   public XtextGraph transform(XtextGraph xtextGraph) {
-    xtextGraph.getEntityExtensionsByNamespace().keySet().stream()
-        .flatMap(namespace -> createEntityMergingContexts(namespace, xtextGraph))
-        .peek(this::validateBaseExtensionCompatibility)
-        .map(entityTypeMerger::mergeIntoBaseType)
-        .forEach(Federation2PureGraphQLUtil::makeAsPureGraphQL);
+      List<TypeDefinition> baseEntityTypes = xtextGraph.getEntityExtensionsByNamespace().keySet().stream()
+              .flatMap(namespace -> createEntityMergingContexts(namespace, xtextGraph))
+              .peek(this::validateBaseExtensionCompatibility)
+              .map(entityTypeMerger::mergeIntoBaseType)
+              .collect(Collectors.toList());
+
+      baseEntityTypes.forEach(Federation2PureGraphQLUtil::makeAsPureGraphQL);
```

Comment:
curious to know how these two things behave differently. Intent wise doesn't look like a change to me